include/odp/api/spec/packet.h

```diff
@@ -237,6 +237,34 @@ typedef struct odp_packet_data_range {
 
 } odp_packet_data_range_t;
 
+/**
+ * Reassembly status in packet
+ */
+typedef enum odp_packet_reassem_status_t {
+	/** Reassembly was not attempted */
+	ODP_PACKET_REASSEM_SKIPPED = 0,
+
+	/** Reassembly was attempted but is incomplete. Multiple distinct
+	  * fragments are available in the packet. The fragments are recovered
+	  * using ``odp_packet_recover_frags()``. */
+	ODP_PACKET_REASSEM_INCOMPLETE,
+
+	/** Reassembly is successfully done and the fragments are delivered as a
+	  * single chained (i.e. segmented) opd_packet_t. */
+	ODP_PACKET_REASSEM_COMPLETE,
+} odp_packet_reassem_status_t;
+
+/**
+ * Result from odp_packet_recover_frags()
+ */
+typedef struct odp_packet_reassem_recover_res_t {
+	/** Number of frags recovered */
+	uint16_t nb_frags;
+
+	/** Time spent in pktio input waiting for reassembly */
+	uint64_t wait_time;
+} odp_packet_reassem_recover_res_t;
```

Comment:
Maybe this could be named odp_reass_packet_result_t.

nb_frags should be num_frags for consistency with other parts of the API.

Wait time lacks specification of time units as discussed today. It probably should be nanoseconds as that is what is used in other ODP APIs.

Reply:
Agreed on all 3 items.